homeassistant/components/yardian/switch.py

```diff
@@ -0,0 +1,73 @@
+"""Support for Yardian integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DEFAULT_WATERING_DURATION, DOMAIN
+from .coordinator import YardianUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entry for a Yardian irrigation switches."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        YardianSwitch(
+            coordinator,
+            i,
+        )
+        for i in range(len(coordinator.data.zones))
+    )
+
+
+class YardianSwitch(CoordinatorEntity[YardianUpdateCoordinator], SwitchEntity):
+    """Representation of a Yardian switch."""
+
+    def __init__(self, coordinator: YardianUpdateCoordinator, zone_id) -> None:
+        """Initialize a Yardian Switch Device."""
+        super().__init__(coordinator)
+        self._zone_id = zone_id
+        self._attr_icon = "mdi:water"
```

Comment:
The static values can be class variables so they don't have to be initialized in init for every switch